sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -496,7 +496,10 @@ private[hive] class TestHiveSparkSession(
   def getLoadedTables: collection.mutable.HashSet[String] = sharedState.loadedTables
 
   def loadTestTable(name: String): Unit = {
-    if (!sharedState.loadedTables.contains(name)) {
+    // LOAD DATA does not work on temporary views. Since temporary views are resolved first,
+    // skip loading if there exists a temporary view with the given name.
+    if (sessionState.catalog.getTempView(name).isEmpty &&
+        !sharedState.loadedTables.contains(name)) {
       // Marks the table as loaded first to prevent infinite mutually recursive table loading.
```

Comment:
Note that this `def` is called as follows:
https://github.com/apache/spark/blob/93ad26be01a47fb075310a26188e238d55110098/sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala#L611-L619

Before this PR, it was resolved to a table in the current database, but after this PR, it will be resolved to a temporary view (if exists).